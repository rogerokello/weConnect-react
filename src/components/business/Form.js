import React, { Component } from 'react';

export const Form = ({id, submitForm, handleEdit="", currentBusiness, formType="Add"}) => {
    console.log("Current Business",submitForm);
    return (
        <form className="form-horizontal" id={id} onSubmit = {submitForm}>
                                    
            <div className="form-group">
                <label className="control-label col-sm-2" htmlFor="businessname">Name:</label>
                <div className="col-sm-10">
                    {formType==="Edit"?
                        <input className="form-control" value={currentBusiness.name} onChange={handleEdit} name="name" placeholder="Business Name" id="businessname" maxLength="100" size="100"/>:
                        <input className="form-control" name="name" placeholder="Business Name" id="businessname" maxLength="100" size="100"/>
                    }
                </div>
            </div>
            
            <div className="form-group">
                <label className="control-label col-sm-2" htmlFor="businesslocation">Location:</label>
                <div className="col-sm-10">
                    {formType==="Edit"?
                        <input className="form-control" value={currentBusiness.location} onChange={handleEdit} name="location" placeholder="Business Location" id="businesslocation" maxLength="100" size="100"/>
                        :
                        <input className="form-control" name="location" placeholder="Business Location" id="businesslocation" maxLength="100" size="100"/>
                    }
                </div>
            </div>
            
            <div className="form-group">
                <label className="control-label col-sm-2" htmlFor="businesscategory">Category:</label>
                <div className="col-sm-10">
                    {formType==="Edit"?
                        <input className="form-control" value={currentBusiness.category} onChange={handleEdit} name="category" placeholder="Business Category" id="businesscategory" maxLength="100" size="100"/>
                        :
                        <input className="form-control" name="category" placeholder="Business Category" id="businesscategory" maxLength="100" size="100"/>
                    }
                </div>
            </div>
            {formType==="Edit"?
                <button id="edit-button" type="submit" className="btn btn-warning text-center"><span className="glyphicon glyphicon-edit"></span> Edit</button>:
                <button id="add-button" type="submit" className="btn btn-success text-center"><span className="glyphicon glyphicon-plus"></span> Register</button>
            }
            
        </form>
    );
}