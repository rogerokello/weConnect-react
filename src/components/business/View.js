import React, { Component } from "react";
import {withRouter} from "react-router-dom";
import {connect} from 'react-redux';
import {PropTypes} from "prop-types";

import Navbar from "../NavigationBar";
import {
    deleteBusiness,
    getAllBusiness,
    getOneBusiness
} from "../../actions/businessActions";

import {Footer} from "../layout/common/Footer";
import { SingleBusiness } from "./layout/SingleBusiness";


export class Viewbusiness extends Component {

    componentWillMount(){
        if(localStorage.getItem("access_token") === null){
          this.props.history.push("/login");
        }
    
        //Dispatch action to get all the businesses
        this.props.getAllBusiness();
    }

    deleteBusiness(id) {
		if (window.confirm('Are you sure you want to delete this business?')) {
            this.props.deleteBusiness(id);
            this.props.history.push("/businesslist");
		}
	}

	render() {
		return (		
			<div>	
				<Navbar/>

                <SingleBusiness 
                    name={this.props.currentBusiness.name}
                    location={this.props.currentBusiness.location}
                    category={this.props.currentBusiness.category}
                />
                
                <Footer/>
			</div>
		);
	}
}

// Find current business based on ID passed in URL
function findCurrentBusiness(businesses, id = -1) {

    // Find business for given id
    return businesses.find(business => {
        return parseInt(business.id, 10) === parseInt(id, 10);
    });
}

Viewbusiness.propTypes = {
	currentBusiness:PropTypes.object
}

const mapStateToProps = (state, ownProps) => {
	let currentBusiness;
	if(state){
		currentBusiness = state.business.businesses.message.length ? findCurrentBusiness(state.business.businesses.message, ownProps.match.params.id) : null;
	}
	return {
        currentBusiness,
    };
};

export default withRouter(connect(mapStateToProps, {deleteBusiness, getAllBusiness, getOneBusiness})(Viewbusiness));