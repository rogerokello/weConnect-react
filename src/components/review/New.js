import React, { Component } from "react";
import {withRouter} from "react-router-dom";
import {connect} from 'react-redux';
import {PropTypes} from "prop-types";
import Loader from 'react-loader-spinner';

import Navbar from "../NavigationBar";
import {addReview} from "../../actions/reviewActions";
import store from "../../store";
import {clearReviewMessages} from "../../actions/actionCreators";
import {loaderPosition} from "../styles/loaderPosition";
import {FetchData} from "../loaders/FetchData";

import {Footer} from "../layout/common/Footer";

export class Newreview extends Component {
	constructor(props) {
        super(props);
        this.submitForm = this.submitForm.bind(this);
    }

	componentWillReceiveProps(recieved){
		if(recieved && recieved.review.status==="success"){
            store.dispatch(clearReviewMessages())
            this.props.history.push(
                {
                    pathname: "/reviews/"+recieved.businessId+"/"+recieved.currentBusiness.name,
                    state: { businessId: recieved.businessId }
                }
            );

		}
	};

    componentWillMount(){
        if(localStorage.getItem("access_token") === null){
            this.props.history.push("/login");
        }
	}

	circularStringify = (object) =>{
		let simpleObj={};

		for (let prop in object){
				if (!object.hasOwnProperty(prop)){
						continue;
				}
				if (typeof(object[prop]) === 'object'){
						continue;
				}
				simpleObj[prop] = object[prop];
		}

		return JSON.stringify(simpleObj);

	}

	
	submitForm = (e) => {
        e.preventDefault();
        
        let reviewSummary = e.target.elements["reviewsummary"].value;
        let reviewDetails = e.target.elements["reviewdetails"].value;

        let ratingDropdown = e.target.elements["rating"];
        let index = ratingDropdown.selectedIndex;
        let businessRating = ratingDropdown.options[index].value;

		let details = {
            star_rating: businessRating,
            review_description: reviewDetails,
            review_summary: reviewSummary,
        };

		this.props.addReview(this.circularStringify(details), this.props.businessId);
	}

	render() {
        const loaderPosition = {
			"position": "absolute",
    		"left": "0%",
    		"top": "10%",
    		"width": "100%"
		}
		return (
			
			<div>
				
				<Navbar/>

                <div className="container-fluid text-center">
                    <FetchData style={loaderPosition} pendingTask={this.props.pendingTask} />    
                    <div className="row content">
                        <h2 className="text-center">
                            Create a review for:{' '}{this.props.currentBusiness.name}
                        </h2>

                        <hr/>

                        <div className="col-sm-5">
                            <h3 className="text-center">Business Details</h3>
                            <div className="list-group">
                                <ul>
                                    <li className="list-group-item active">
                                        <h4 className="list-group-item-heading">
                                            Business Information:{' '} {this.props.currentBusiness.name}
                                        </h4>
                                    </li>
                                    <li  className="list-group-item text-left">
                                        <div>
                                            <span className="list-group-item-heading">
                                                <strong>Name:</strong>
                                            </span>
                                            <span className="list-group-item-text">
                                                {' '}{this.props.currentBusiness.name}
                                            </span>
                                        </div>
                                        <div>
                                            <span className="list-group-item-heading">
                                                <strong>Location:</strong>
                                            </span>
                                            <span className="list-group-item-text">
                                                {' '}{this.props.currentBusiness.location}
                                            </span>
                                        </div>
                                        <div>
                                            <span className="list-group-item-heading">
                                                <strong>Category:</strong>
                                            </span>
                                            <span className="list-group-item-text">
                                                {' '}{this.props.currentBusiness.category}
                                            </span>
                                        </div>
                                    </li>
                            </ul>
                        </div>  
                    </div>
                        
                    <div className="col-sm-6">
                        <h3 className="text-center">Your review	</h3>
                        <form id="reviewform" className="form-horizontal" onSubmit={this.submitForm}>
                            <div className="form-group">
                                <label className="control-label col-sm-4" htmlFor="businessrating">Rate(<span className="glyphicon glyphicon-star"></span>):</label>
                                <div className="col-sm-8">
                                    <select name="rating" className="form-control" id="businessrating">
                                        <option>5</option>
                                        <option>4</option>
                                        <option>3</option>
                                        <option>2</option>
                                        <option>1</option>
                                    </select>
                                </div>
                            </div>
                            <div className="form-group">
                                <label className="control-label col-sm-4" htmlFor="reviewsummary">Review Summary:</label>
                                <div className="col-sm-8">
                                <input className="form-control" name="reviewsummary" placeholder="Review Summary" id="reviewsummary" maxLength="50" size="50"/>
                                </div>
                            </div>
                            <div className="form-group">
                                <label className="control-label col-sm-4" htmlFor="review">Your review:</label>
                                <div className="col-sm-8">
                                <textarea name="reviewdetails" className="form-control" rows="5" id="review"></textarea>
                                </div>
                            </div>
                            <button type="submit" className="btn btn-warning text-center"><span className="glyphicon glyphicon-plus"></span> Add your review</button>
                        </form>
                    </div>
                    </div>
                </div>

                <Footer/>
                
			</div>
		);
	}
}

Newreview.propTypes = {
	addReview: PropTypes.func,
	review: PropTypes.object
}

// Find current business based on ID passed in URL
function findCurrentBusiness(businesses, id = -1) {

    // Find business for given id
    return businesses.find(business => {
        return parseInt(business.id, 10) === parseInt(id, 10);
    });
}

const mapStateToProps = (state, ownProps) => {
    let currentBusiness;

	if(state){
		currentBusiness = state.business.businesses.message.length ? findCurrentBusiness(state.business.businesses.message, ownProps.match.params.id) : null;
    }
    
    return {
        currentBusiness: currentBusiness, 
        review: state.review.reviewMessage,
        businessId: ownProps.match.params.id,
        pendingTask: state.pendingTasksReducer
    }
};


export default withRouter(connect(mapStateToProps, {addReview})(Newreview));