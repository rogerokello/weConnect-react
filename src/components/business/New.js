import React, { Component } from "react";
import {withRouter} from "react-router-dom";
import {connect} from 'react-redux';
import {PropTypes} from "prop-types";
import {NotificationManager} from 'react-notifications';
import Loader from 'react-loader-spinner';

import Navbar from "../NavigationBar";
import {clearNewBusinessMessageStatus} from "../../actions/actionCreators";
import {loaderPosition} from "../styles/loaderPosition";
import {addBusiness} from "../../actions/businessActions";
import store from "../../store";

import {stringify} from "../../helpers/circularstringify";

import {Footer} from "../../components/layout/common/Footer";

import {FetchData} from "../loaders/FetchData";

import {Form} from "./Form";

import {Business} from "./layout/Business";

export class Newbusiness extends Component {
	constructor(props) {
        super(props);
        this.submitForm = this.submitForm.bind(this);
    }

	componentWillReceiveProps(recieved){
		if(recieved && recieved.newBusinessMessage.status==="success"){
			this.props.history.push("/businesslist");
			NotificationManager.info(recieved.newBusinessMessage.message, "", 5000);
			store.dispatch(clearNewBusinessMessageStatus())
		}

		if(recieved && recieved.newBusinessMessage.status === "failure"){
			NotificationManager.info(recieved.newBusinessMessage.message, "", 5000);
			store.dispatch(clearNewBusinessMessageStatus())
		}
	};

    componentWillMount(){
        if(localStorage.getItem("access_token") === null){
            this.props.history.push("/login");
        }
	}

	submitForm = (e) => {
		e.preventDefault();

		let details = {
            name: e.target.elements.name.value,
            category: e.target.elements.category.value,
            location: e.target.elements.location.value,
        };

		this.props.addBusiness(stringify(details))
	}

	render() {

		const loaderPosition = {
			"position": "absolute",
    		"left": "0%",
    		"top": "50%",
    		"width": "100%"
		}

		return (
			
			<div>
				
				<Navbar/>
				<Business
					header={"Register a new business"}
					fetchDataLoader={<FetchData style={loaderPosition} pendingTask={this.props.pendingTask} />}
					form={
						<Form	id={"newbusinessForm"} 
								submitForm={this.submitForm.bind(this)}
								currentBusiness={this.props.currentBusiness}
						/>
					}
				/>
				<Footer />
			</div>
		);
	}
}

Newbusiness.propTypes = {
	addBusiness: PropTypes.func.isRequired,
	business:PropTypes.object
}

const mapStateToProps = state => {
    return {
		newBusinessMessage: state.business.newbusinessMessage,
		pendingTask: state.pendingTasksReducer,
    }
};


export default withRouter(connect(mapStateToProps, {addBusiness})(Newbusiness));