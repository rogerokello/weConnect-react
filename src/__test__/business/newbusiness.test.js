import React from 'react';
import expect from 'expect';
import {MemoryRouter} from 'react-router-dom';
import {shallow ,mount} from 'enzyme';
import configureStore from 'redux-mock-store';
import {Provider} from "react-redux";


import mainReducer from "../../reducers";
import {Newbusiness} from '../../components/business/New';

const middlewares = []
const mockStore = configureStore(mainReducer,middlewares)

const localStorageMock = {
    getItem: jest.fn(),
    setItem: jest.fn(),
    clear: jest.fn()
};

global.localStorage = localStorageMock;

describe ('New business Page', () => {
    const initialState = {

    }
    let wrapper;
    let store = mockStore(initialState)
    let mockNewbusinessfn = jest.fn();
    beforeEach( () => {
        
        wrapper = mount(
            <Provider store={store}>
                <MemoryRouter>
                    <Newbusiness addBusiness={mockNewbusinessfn} />
                </MemoryRouter>
            </Provider>
        );
    });
    
    it('checks register button can work', () => {

        wrapper.find('#newbusinessForm').at(0).simulate(
            'submit', 
            {preventDefault() {}}
        )
        expect(mockNewbusinessfn.mock.calls.length).toBe(1)
    });
    it('should match snapshot',() =>{
        const wrapper = shallow(<Newbusiness addBusiness={mockNewbusinessfn} />)
        expect(wrapper).toMatchSnapshot();
    })

});