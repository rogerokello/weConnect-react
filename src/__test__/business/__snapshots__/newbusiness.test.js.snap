// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New business Page should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Newbusiness
    addBusiness={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(Connect(Navbar)) />,
        <Unknown
          fetchDataLoader={
                    <Unknown
                              pendingTask={undefined}
                              style={
                                        Object {
                                                  "left": "0%",
                                                  "position": "absolute",
                                                  "top": "50%",
                                                  "width": "100%",
                                                }
                              }
                    />
          }
          form={
                    <Unknown
                              currentBusiness={undefined}
                              id="newbusinessForm"
                              submitForm={[Function]}
                    />
          }
          header="Register a new business"
/>,
        <Unknown />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "fetchDataLoader": <Unknown
            pendingTask={undefined}
            style={
                        Object {
                                    "left": "0%",
                                    "position": "absolute",
                                    "top": "50%",
                                    "width": "100%",
                                  }
            }
/>,
          "form": <Unknown
            currentBusiness={undefined}
            id="newbusinessForm"
            submitForm={[Function]}
/>,
          "header": "Register a new business",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(Connect(Navbar)) />,
          <Unknown
            fetchDataLoader={
                        <Unknown
                                    pendingTask={undefined}
                                    style={
                                                Object {
                                                            "left": "0%",
                                                            "position": "absolute",
                                                            "top": "50%",
                                                            "width": "100%",
                                                          }
                                    }
                        />
            }
            form={
                        <Unknown
                                    currentBusiness={undefined}
                                    id="newbusinessForm"
                                    submitForm={[Function]}
                        />
            }
            header="Register a new business"
/>,
          <Unknown />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "fetchDataLoader": <Unknown
              pendingTask={undefined}
              style={
                            Object {
                                          "left": "0%",
                                          "position": "absolute",
                                          "top": "50%",
                                          "width": "100%",
                                        }
              }
/>,
            "form": <Unknown
              currentBusiness={undefined}
              id="newbusinessForm"
              submitForm={[Function]}
/>,
            "header": "Register a new business",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
